src/client/providers/execInTerminalProvider.ts

```diff
@@ -0,0 +1,35 @@
+"use strict";
+import * as vscode from "vscode";
+import * as settings from "./../common/configSettings";
+
+export function activateExecInTerminalProvider() {
+    vscode.commands.registerCommand("python.execInTerminal", execInTerminal);
+}
+
+function execInTerminal(filePath: string) {
+    const currentPythonPath = settings.PythonSettings.getInstance().pythonPath;
+
+    console.log(filePath);
+    if (filePath === undefined) {
+        const activeEditor = vscode.window.activeTextEditor;
+        if (activeEditor !== undefined) {
+            if (!activeEditor.document.isUntitled) {
+                filePath = activeEditor.document.fileName;
```

Comment:
@jabooth  You might want to check whether the current file is a python file.
I believe you can inspect the language of the active editor.